src/main/java/cs2103/v15_1j/jimjim/ui/SearchPaneController.java

```diff
@@ -52,26 +54,26 @@ private void setUpSearchPane(){
 		searchScrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
 	}
 
-	public void refreshData(DataLists lists){
-		this.lists = lists;
+	public void refreshData(DataLists masterList, DataLists searchResultsList){
+		this.masterList = masterList;
+		this.searchResultsList = searchResultsList;
 		showSearchResults();
 	}
 
 	private void showSearchResults(){
 		searchGridPane.getChildren().clear();
-		int counter = 0;
 		int rowNo = 0;
 
 		Label titleLabel = new Label("Search Results");
 		searchGridPane.add(titleLabel, 0, rowNo, 4, 1);
 
-		for(Event event: lists.getEventsList()){
-			counter++;
+		for(Event event: searchResultsList.getEventsList()){
 			Circle dot = new Circle(3.0, Color.RED);
 			GridPane.setHalignment(dot, HPos.CENTER);
 			searchGridPane.addColumn(0, dot);
 
-			Label idLabel = new Label("[E"+counter+"]");
+			int id = masterList.getEventsList().indexOf(event) + 1;
```

Comment:
no need. just use `masterList.indexOf` it will automatically search the right list based on the class